lib/kafka_ex/new/adapter.ex

```diff
@@ -384,32 +392,77 @@ defmodule KafkaEx.New.Adapter do
     ]
   end
 
-  def offset_commit_request(request, client_consumer_group) do
-    consumer_group = request.consumer_group || client_consumer_group
-
-    {%Kayrock.OffsetCommit.V0.Request{
-       group_id: consumer_group,
-       topics: [
-         %{
-           topic: request.topic,
-           partitions: [
-             %{
-               partition: request.partition,
-               offset: request.offset,
-               metadata: ""
-             }
-           ]
-         }
-       ]
-     }, consumer_group}
+  def offset_commit_request(offset_commit_request, client_consumer_group) do
+    consumer_group =
+      offset_commit_request.consumer_group || client_consumer_group
+
+    request =
+      Kayrock.OffsetCommit.get_request_struct(offset_commit_request.api_version)
+
+    request = %{
+      request
+      | group_id: consumer_group,
+        topics: [
+          %{
+            topic: offset_commit_request.topic,
+            partitions: [
+              %{
+                partition: offset_commit_request.partition,
+                offset: offset_commit_request.offset,
+                metadata: ""
+              }
+            ]
+          }
+        ]
+    }
+
+    request =
```

Comment:
This request is from Kayrock.OffsetCommit.get_request_struct/1, which we pass the api version into. It seems like it should handle this case statement as well, otherwise it seems like there's no point in actually passing in the api version?

Reply:
I'm not sure I understand your question.  I'll try to answer, but not 100% sure it'll be an answer to what you're actually asking.

We have to get the request struct from Kayrock so that it's the right struct.  Each version of each message is a different struct - e.g., `%Kayrock.OffsetCommit.V0.Request{}`.  The case statement here (which, yeah, is kind of ugly... but I didn't see any simple way to clean it up significantly without losing readability) populates fields of the struct depending on what version it is (since different versions can have different fields).  Kayrock is sort of intentionally not concerned with knowing how to populate the fields - it just knows what fields should be in what structs and how to serialize/deserialize them.

The weird/stupid thing is that the OffsetCommit request had a timestamp field in v1 that got removed in v2.  Otherwise this would probably be as simple as checking if version >= 1 and populating the generation_id and member_id fields which are common in v >= 1.

Reply:
Gotcha, makes sense.